### DISCUSSION TITLE
Filter pull requests by author

### DISCUSSION THREAD

**Discussion by simon-fermor** [community/community#24810] [Apps, API and Webhooks]
<p>I’d like to get details of pull requests from a repository by a given author / user.  I’d like to use an argument for author in the query shown below.  Do I need to use a search filter?  I’m using the Explorer and I don’t see which arguments can be used for particular fields or the syntax and arguments for the search field.</p>
<pre><code>query ($number_of_pull_requests: Int!) {
  repository(owner: "myorganization", name: "reponame") {
    pullRequests(first: $number_of_pull_requests, orderBy: {field: CREATED_AT, direction: DESC}) {
      edges {
        cursor
        node {
          id
          closedAt
          number
          author {
            login
          }
        }
      }
    }
  }
}
</code></pre>

**Comment by parithiban** [comment, marked as answer]
<p><a class="mention" href="/simon-at-timeshift">@simon-at-timeshift</a> </p>
<p>Yes you can try it via a search query</p>
<pre><code>query searchRepos($qstr: String!, $first: Int!) {
  search(query: $qstr, type: ISSUE, first: $first) {
    nodes {
      __typename
      ... on PullRequest {
        url
        number
        author {
          login
        }
      }
    }
  }
}

{
    "qstr": "repo:parithiban/git-slack-bot type:pr is:open author:parithiban",
    "first": 5
}
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/07d108cbcbaa1c81040d16103575f52d477a1ad3.png" data-download-href="/uploads/short-url/179caK61KMsjEGQgUwzoKVTLULV.png?dl=1" title="Screen Shot 2019-11-30 at 7.12.29 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7281166/181086966-54a503c8-e2c1-448c-b600-35fcd4bcc85a.png" alt="Screen Shot 2019-11-30 at 7.12.29 PM.png" data-base62-sha1="179caK61KMsjEGQgUwzoKVTLULV" width="690" height="377" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/07d108cbcbaa1c81040d16103575f52d477a1ad3_2_690x377.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/07d108cbcbaa1c81040d16103575f52d477a1ad3_2_1035x565.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/0/07d108cbcbaa1c81040d16103575f52d477a1ad3_2_1380x754.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/07d108cbcbaa1c81040d16103575f52d477a1ad3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-11-30 at 7.12.29 PM.png</span><span class="informations">2872×1572 354 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
